cpp/src/arrow/array.h

```diff
@@ -170,24 +170,34 @@ struct ARROW_EXPORT ArrayData {
 
   // Access a buffer's data as a typed C pointer
   template <typename T>
-  inline const T* GetValues(int i) const {
+  inline const T* GetValues(int i, int64_t offset_) const {
```

Comment:
The `foo_` naming is reserved for class members

Reply:
@pitrou since this function can be used to access any of the buffers, the "GetValues" name could be misleading. Perhaps `GetBufferAs<T>(i)` or something?